packages/mobile/src/account/saga.ts

```diff
@@ -156,6 +173,66 @@ export function* watchKycStatus() {
   }
 }
 
+export function* generateSignedMessage() {
+  try {
+    const wallet: UnlockableWallet = yield call(getWallet)
+    const address: string = yield select(walletAddressSelector)
+    yield call(unlockAccount, address)
+
+    const signedMessage = yield call(
+      [wallet, 'signPersonalMessage'],
+      address,
+      ensureLeading0x(Buffer.from('valora auth message').toString('hex'))
+    )
+    yield call(storeSignedMessage, signedMessage)
+    yield put(saveSignedMessage())
+  } catch (error) {
+    throw error
+  }
+}
+
+export function* handleUpdateAccountRegistration(extraProperties: RegistrationProperties = {}) {
+  const signedMessage = yield call(retrieveSignedMessage)
+  if (!signedMessage) {
+    // ensures backwards compatibility - this should happen only for updating the
+    // fcm token when an existing user updates the app and the signed message is
+    // not yet generated
+    Logger.warn(
+      `${TAG}@handleUpdateAccountRegistration`,
+      'Tried to update account registration without signed message'
+    )
+    return
+  }
+
+  const address = yield select(walletAddressSelector)
+  const appVersion = DeviceInfo.getVersion()
+  const language = yield select(currentLanguageSelector)
+  const country = yield select(userLocationDataSelector)
+
+  let fcmToken
+  try {
+    fcmToken = yield call([firebase.app().messaging(), 'getToken'])
+  } catch (error) {
+    Logger.error(`${TAG}@handleUpdateAccountRegistration`, 'Could not get fcm token', error)
```

Comment:
oh yeah this is a great point. i updated the implementation to only update defined values, the endpoint does this check already but i think it's more correct for the app to do this rather than rely on an endpoint implementation that might change.